lua/telescope/config.lua

```diff
@@ -34,131 +58,273 @@ function config.set_pickers(pickers)
+    preview_cutoff = 120,
+  },
+
+  vertical = {
+    preview_cutoff = 40,
+  },
+
+  center = {
+    preview_cutoff = 40,
+  },
+}
+
+local layout_config_description = string.format([[
+    Determines the default configuration values for layout strategies.
+    See |telescope.layout| for details of the configurations options for
+    each strategy.
+
+    Allows setting defaults for all strategies as top level options and
+    for overriding for specific options.
+    For example, the default values below set the default width to 80%% of
+    the screen width for all strategies except 'center', which has width
+    of 50%% of the screen width.
+
+    Default: %s
+]], vim.inspect(
+  layout_config_defaults,
+  { newline = "\n    ", indent = "  " }
+))
+
+-- A table of all the usual defaults for telescope.
+-- Keys will be the name of the default,
+-- values will be a list where:
+-- - first entry is the value
+-- - second entry is the description of the option
+
+local telescope_defaults = {
+
+  sorting_strategy = {
+    "descending",
+    [[
     Determines the direction "better" results are sorted towards.
 
     Available options are:
     - "descending" (default)
-    - "ascending"]])
+    - "ascending"]],
+  },
 
-  set("selection_strategy", "reset", [[
+  selection_strategy = {
+    "reset",
+    [[
     Determines how the cursor acts after each sort iteration.
 
     Available options are:
     - "reset" (default)
     - "follow"
-    - "row"]])
+    - "row"]],
+  },
 
-  set("scroll_strategy", "cycle", [[
+  scroll_strategy = {
+    "cycle",
+    [[
     Determines what happens you try to scroll past view of the picker.
 
     Available options are:
     - "cycle" (default)
-    - "limit"]])
+    - "limit"]],
+  },
+
+  layout_strategy = {
+    "horizontal",
+    [[
+    Determines the default layout of Telescope pickers.
+    See |telescope.layout| for details of the available strategies.
 
-  set("layout_strategy", "horizontal")
-  set("layout_defaults", {})
+    Default: 'horizontal']],
+  },
 
-  set("width", 0.75)
-  set("winblend", 0)
-  set("prompt_position", "bottom")
-  set("preview_cutoff", 120)
+  layout_config = { layout_config_defaults, layout_config_description },
 
-  set("results_height", 1)
-  set("results_width", 0.8)
+  winblend = { 0 },
 
-  set("prompt_prefix", "> ", [[
+  prompt_prefix = { "> ", [[
     Will be shown in front of the prompt.
 
-    Default: '> ']])
-  set("selection_caret", "> ", [[
+    Default: '> ']]
+  },
+
+  selection_caret = { "> ", [[
     Will be shown in front of the selection.
 
-    Default: '> ']])
-  set("entry_prefix", "  ", [[
+    Default: '> ']]
+  },
+
+  entry_prefix = { "  ", [[
     Prefix in front of each result entry. Current selection not included.
 
-    Default: '  ']])
-  set("initial_mode", "insert")
+    Default: '  ']]
+  },
+
+  initial_mode = { "insert" },
 
-  set("border", {})
-  set("borderchars", { '─', '│', '─', '│', '╭', '╮', '╯', '╰'})
+  border = { {} },
```

Comment:
Are we doing anything with that. I am currently confused by that.

If this thing is nil we are not calling the local `get_border_size` in layout_strategies.lua function and if i am using a theme (dropdown) its true, so `get_border_size` will be called. I can set it to false, obv and that works as expected but maybe we should set it to true on default. And give it a description. Thoughts?!

Reply:
I think that this should probably be just a boolean.

I've changed the default to `true` and added some documentation.
I'm happy with the situation now, but will wait for someone else to "Resolve" once they're happy with it 🙂